### DISCUSSION TITLE
Use BAS outputs to simulate 10K temp sensor??

### DISCUSSION THREAD

**Discussion by johnny2678** [tagyoureit/nodejs-poolController#572] [General]
This probably sounds dumb... but hear me out.

I've got an SWG - Pentair IC40.  It's about 7 years old but still generating Cl (~3000 hours according to the onboard LEDs).  Problem is I keep getting the "Low Temp" warning.  

Low Temp, in FL, in the dead heat of summer? 😂🤷‍♂️

Right now my pool temp is 83 and it's only that low because it's been stormy and overcast the past 2 days and I've got the low temp warning.  Onboard LEDs say the pool temp is 46-55F (the 40+60% LEDs are lit when I check temp).

Conventional wisdom says the flow switch 10K temp sensor is bad.  Time to swap it out.  I've done that.  I even hooked up the flow switch to the BAS prior to installing in the SWG, set up a temp sensor and flow switch on open inputs and confirmed that both were reading correctly.  But after installing in the SWG, temp reads 20-30F lower.

Less conventional wisdom says clip the green/white wires going into the SWG from the flow switch and wire in an external 10K sensor that's tapped into the pool plumbing.  I've tried that too - it's how I've got it wired right now.  The 10K sensor was reading correctly prior to wiring it up to the SWG.  No dice, SWG still registers 20-30F lower.

Now, I can get away with this in the summer time because most days the actual pool temp gets to 87-88 degrees, which is enough to push the onboard SWG temp reading to 56-65F which turns off the Low Temp warning.  But in the Fall/Winter/Spring....

I know I'll have to spring for another IC40 eventually but throwing down $1200-1400 for a cell that's otherwise working seems excessive.  So I had the thought, can I trick it to think that water temp is normal by just simulating 100Fand hoping that the SWG reads it as 70-80F?

AFAIK, temp is interpreted by reading KΩ so can I somehow set an output on the BAS to mimic whatever KΩ reading corresponds to 100F?
```
0-10V Outputs:
Minimum Output Load: 1KΩ
Resolution: tbd
Full scale linearity: 0.1%
```

Not too concerned about water temp falling below 55 here.

Just a thought that I wanted to get some feedback on.  Hope it makes sense.

**Comment by rstrouse** [comment]
You might just be better off putting a 4k resistor in its place.  This tells me that either the internal software went out of calibration or there is a voltage drop on the sensor wire and the ADC is reading higher resistance.  I can't quite put my finger on the cold water thingy since older cells had no concept of this.  I'm not so sure that the issue isn't that the temp compensation on the EC sensor inside the cell is not very good so they opted for a too cold approach.  When the water temp is actually low it does read salt levels < 2650 which is essentially 0 to the cell.

**Reply by johnny2678** [reply]
hmm so just wire a 4k resistor between the green and white wires and remove the flow switch temp wires? (leaving the flow switch black/red wires connected) 

Sorry - opted to go into technology so I wouldn't have to take EE 😂

![image](https://user-images.githubusercontent.com/8219743/184669234-a97bead8-3a3d-4aff-87e2-fc392ba11090.png)

**Reply by tagyoureit** [reply]
Just another thought, have you given your cell a good acid bath recently?  Maybe there is some buildup that is interfering with the readings.

**Reply by johnny2678** [reply]
yes, @tagyoureit - meant to mention that in my initial post.  I have my acid injector before my SWG in plumbing so scale doesn't get a chance to build up.  Even so, I gave it a bath about a month ago, even though plates were clean.

**Reply by rstrouse** [reply]
Yes on the cell side such that it thinks there is 4k resistance between the green/white wires.   Normally this would make the cell think that the water is 117F but if the reference voltage is low on the the ADC it will translate this into a lower temperature but within the range of the cell.  There is no guarantee that the issue is bad reference resistance on the cell though.  All those circuits inside the cell and are encased in the same carbonite that Han Solo was encased in.  You need to wear a couple of sticky buns on the side of your head to get through that stuff.

**Comment by johnny2678** [comment]
ok, is there some scenario where I can change the SWG setpoint from Screenlogic but not from NJSPC?  But I can still change other pool settings, pump speed, spa mode, heater, etc from NJSPC?

Grabbed a quick replay with the following steps:

1.  set SWG to 20%
2. start capture
3. from screenlogic, change pool setpoint to 40%
4. observe additional LED lit (was just one 20% LED, now two - SUPER blurry pics from a zoomed in Pad Cam bc I'm too lazy to keep running out there 😂
![image](https://user-images.githubusercontent.com/8219743/184996249-b774a097-f6cb-43b3-8e1c-700c34e826d5.png)
![image](https://user-images.githubusercontent.com/8219743/184996262-a75c33aa-7e3d-47c4-a3c0-23a2e06ef76a.png)

5. Change SWG Pool Setpoint to 60% in dashpanel
6. Confirm setting in dash
![image](https://user-images.githubusercontent.com/8219743/184996363-6b249555-1929-448b-9a02-e89e05c5eabc.png)

7. However, on SWG, only 2 LEDs remain lit and screenlogic still reads 40% output
![image](https://user-images.githubusercontent.com/8219743/184996535-f06b2015-afb0-4bf6-846a-ff3ccca8a141.png)


Admittedly, I have been playing with the temp sensor on the flow switch per the thread above.  I wired the flow switch 10K back in and it is reading temp properly... for now.  The 4th and 5th LEDs are lit when I do diagnostics which tells me it's reading between 86-95 which is accurate.  I have my doubts that temp will keep reading accurately but we'll see.

But it seems i've lost SWG control from NJSPC in the process?

Hoping you guys that live in the matrix can spot some dumb thing I'm overlooking
[replay (31).zip](https://github.com/tagyoureit/nodejs-poolController/files/9355062/replay.31.zip)

**Reply by tagyoureit** [reply]
This is strange.  I see you requesting to set the chlor to 20 and it says, ya, sure, I'm a 60.
![2022-08-16_17-33-36](https://user-images.githubusercontent.com/1646840/185008300-755f484a-8cc8-48df-af6e-fa106256a426.png)

And then you ask it to go to 40, and it says, NO WAY, I'm still staying at 60.
![2022-08-16_17-33-49](https://user-images.githubusercontent.com/1646840/185008349-2849cd50-61c7-47c0-8cc0-36c46e9d5021.png)

The chlor might think it is under local control and ignoring your requests.  You could either/both do a hard power reset on the chlor (aka turn off the pool circuit if that kills power) or try to restart njsPC (I didn't see any of these packets sent to the chlor).

**Reply by johnny2678** [reply]
> And then you ask it to go to 40, and it says, NO WAY, I'm still staying at 60.

so, the request to go to 40 came from screenlogic.  odd thing is... that took.  the swg leds lit up to 40%. but njspc still says 60%

screenlogic and njspc both get to the OCP/swg over the same bus and comm wires so im confused.

tried pulling the power to the swg and restarting again this AM but same results.  Anything else i should be looking at? I'm befuddled

edit: interestingly enough, the SWG flow sensor is still reading the correct temp this AM and SWG status is `Ok`

**Reply by tagyoureit** [reply]
First off, duh.  The 60 in the response is the salt level (60*50) or 3000.  And the 128 is just the status bit.  So red herring on my part and not referencing my docs/code.  

The first one is the response to SL.  The second is the OCP setting the level to 40%.  

I'm not seeing njsPC send out any 153 messages to tell the OCP to change the level.  Can you watch message manage and (after init is completed) try to change it and see if you can see 153, 217 or 25 messages?

**Reply by johnny2678** [reply]
sorry, I've been around your project for years but haven't used message manager before and not seeing anything in the wiki search.  Do you have a link?

**Reply by rstrouse** [reply]
Sometimes the simplest explanation is the right one.  First, delete the chlorinator in njsPC and let EasyTouch notify us of its existence by going into the hamburger menu => System tab and pressing Reload Config.  The chlorinator you have defined is defined as an external chlorinator that is being controlled by some other means.

**Reply by johnny2678** [reply]
It didn't come back after reloading config.  Went out to the panel and it says Chlorinator not present.  Oy...

Haven't touched any of the comm wires (only the flow switch wires) but maybe they've shaken loose.  Will have to take some time later today to get into the panel. 

In the mean time, it's just running on manual control.

**Reply by johnny2678** [reply]
times like this I wish I had an extra cell lying around to just check out comms/wiring.  Have long suspected my SWG circuitry was wonky due to all the SWG status errors.  Wouldn't you know it, temp is reading correct this AM from the on cell diagnostics.  It only cost me rs485 comms with the cell 😂

**Reply by rstrouse** [reply]
Ok so lets manually add it through njsPC.  Make sure you select your EasyTouch as the master.  The way you had this configured was as an external chlorinator that is controlled by some other process not Nixie and not EasyTouch.

**Reply by johnny2678** [reply]
will that work even though the OCP can't see the SWG?

**Reply by johnny2678** [reply]
whoa... ok now I have SWG control back through njspc.  I can't get my head around this.  The OCP didn't detect the SWG until I added it in njspc?

**Reply by rstrouse** [reply]
njsPC told EasyTouch the chlorinator was there and it waited until it was verified.   There is a checkbox in EasyTouch buried somewhere that enables the chlorinator.  njsPC just made sure it was clicked.

njsPC is capable of controlling multiple chlorinators which your EasyTouch is not so you can gang them up if you need to using multiple RS485 adapters.  Heck with any of the Pentair controllers you can only install one chlorinator.  This is true even if you have a separate pool  and spa system.

**Comment by johnny2678** [comment]
Just wanted to say that as of this morning, my SWG flow switch is still reporting temp accurately (for now) and I again have control of the SWG via njspc->mqtt->nodered

Not sure what sent the SWG out of whack, don't remember setting it up as external, i think njspc just detected it that way.  Either way, I'm a happy camper this AM.  

Thanks again for the support @rstrouse @tagyoureit

**Reply by johnny2678** [reply]
side note, I did rough in a 4.7K resistor with a 10K temp sensor and read it through the BAS and you were 100% right @rstrouse - temp read around 120F (from 83-84).

If I ever have to use the 10K input to my SWG again I will try that

**Reply by rstrouse** [reply]
Once I thought I was wrong but I was mistaken.
